server/protocols/webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/EventDeadLettersIntegrationTest.java

```diff
@@ -0,0 +1,473 @@
+
+    public class RetryEventsListener implements MailboxListener.GroupMailboxListener {
+        final Group GROUP = new RetryEventsListenerGroup();
+
+        private int retriesBeforeSuccess;
+        private Map<Event.EventId, Integer> retries;
+        private List<Event> successfulEvents;
+
+        private void init() {
+            this.retriesBeforeSuccess = 0;
+            this.retries = new HashMap<>();
+            this.successfulEvents = new ArrayList<>();
+        }
+
+        RetryEventsListener() {
+            init();
+        }
+
+        @Override
+        public Group getDefaultGroup() {
+            return GROUP;
+        }
+
+        @Override
+        public void event(Event event) throws Exception {
+            int currentRetries = extractEventRetries(event).orElse(0);
+
+            if (currentRetries < retriesBeforeSuccess) {
+                retries.put(event.getEventId(), currentRetries + 1);
+                throw new RuntimeException("throw to trigger retry");
+            } else {
+                retries.remove(event.getEventId());
+                successfulEvents.add(event);
+            }
+        }
+
+        private Optional<Integer> extractEventRetries(Event event) {
+            return Optional.ofNullable(retries.get(event.getEventId()));
+        }
+
+        List<Event> getSuccessfulEvents() {
+            return successfulEvents;
+        }
+
+        void setRetriesBeforeSuccess(int retriesBeforeSuccess) {
+            this.retriesBeforeSuccess = retriesBeforeSuccess;
+        }
+
+        void clear() {
+            init();
+        }
+    }
+
+    private static final String DOMAIN = "domain.tld";
+    private static final String BOB = "bob@" + DOMAIN;
+    private static final String BOB_PASSWORD = "bobPassword";
+    private static final String EVENTS_ACTION = "reDeliver";
+    private static final String SERIALIZED_GROUP = new RetryEventsListenerGroup().asString();
+
+    private Duration slowPacedPollInterval = ONE_HUNDRED_MILLISECONDS;
+    private ConditionFactory calmlyAwait = Awaitility.with()
+        .pollInterval(slowPacedPollInterval)
+        .and()
+        .with()
+        .pollDelay(slowPacedPollInterval)
+        .await();
+    private ConditionFactory awaitAtMostTenSeconds = calmlyAwait.atMost(10, TimeUnit.SECONDS);
+    private RetryEventsListener retryEventsListener = new RetryEventsListener();
+
+    @ClassRule
+    public static DockerCassandraRule cassandra = new DockerCassandraRule();
+
+    @Rule
+    public CassandraJmapTestRule cassandraJmapTestRule = CassandraJmapTestRule.defaultTestRule();
+
+    private GuiceJamesServer guiceJamesServer;
+    private MailboxProbeImpl mailboxProbe;
+
+    @Before
+    public void setUp() throws Exception {
+        retryEventsListener.clear();
+        guiceJamesServer = cassandraJmapTestRule.jmapServer(cassandra.getModule())
+            .overrideWith(new WebAdminConfigurationModule())
```

Comment:
why do you need this override?

Reply:
Cause no conf is shipped by default, leading to webAdmin not being started.

Reply:
It's a case where I'd rather kill WebAdminConfigurationModule and replace it with 
```java bind(WebAdminConfiguration.class).toInstance(WebAdminConfiguration.TEST_CONFIGURATION)
```
because it's so much obvious what we can expect from it